utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/encode/AvroSchemaOutputValidator.scala

```diff
@@ -0,0 +1,292 @@
+    def isValueValid(typedWithObject: TypedObjectWithValue, schema: Schema): Boolean = {
+      val enumValue = typedWithObject.value match {
+        case enum: EnumSymbol => Some(`enum`.toString)
+        case str: String => Some(str)
+        case _ => None
+      }
+
+      enumValue.exists(schema.getEnumSymbols.asScala.contains)
+    }
+
+    validateWithValue[java.lang.String](isValueValid, typingResult: TypingResult, schema: Schema, path: Option[String])
+  }
+
+  private def validateFixed(typingResult: TypingResult, schema: Schema, path: Option[String]): Validated[NonEmptyList[OutputValidatorError], Unit] = {
+    def isValueValid(typedWithObject: TypedObjectWithValue, schema: Schema): Boolean = {
+      val fixedStringValue = typedWithObject.value match {
+        case fixed: Fixed => Some(fixed.bytes())
+        case buffer: ByteBuffer => Some(buffer.array())
+        case bytes: Array[Byte] => Some(bytes)
+        case str: String => Some(str.getBytes(StandardCharsets.UTF_8))
+        case _ => None
+      }
+
+      fixedStringValue.exists(_.length == schema.getFixedSize)
+    }
+
+    validateWithValue[java.lang.String](isValueValid, typingResult: TypingResult, schema: Schema, path: Option[String])
+  }
+
+  private def validateUUID(typingResult: TypingResult, schema: Schema, path: Option[String]): Validated[NonEmptyList[OutputValidatorError], Unit] = {
+    def isValueValid(typedWithObject: TypedObjectWithValue, schema: Schema): Boolean = {
+      typedWithObject.value match {
+        case _:UUID => true
+        case str: String => Try(UUID.fromString(str)).toValidated.isValid
+        case _ => false
+      }
+    }
+
+    validateWithValue[java.lang.String](isValueValid, typingResult: TypingResult, schema: Schema, path: Option[String])
+  }
+
+  private def validateWithValue[T:ClassTag](isValueValid: (TypedObjectWithValue, Schema) => Boolean, typingResult: TypingResult, schema: Schema, path: Option[String]) = {
+    val validationResult = validateClass[T](typingResult, schema, path)
+
+    validationResult.andThen{ _ =>
+      typingResult match {
+        case obj: TypedObjectWithValue if isValueValid(obj, schema) => valid
+        case _: TypedObjectWithValue => invalid(typingResult, schema, path)
+        case _ => validationResult
+      }
+    }
+
+  }
+
+  private def validateClass[T:ClassTag](typingResult: TypingResult, schema: Schema, path: Option[String]): Validated[NonEmptyList[OutputValidatorTypeError], Unit] = {
+    val schemaAsTypedResult = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema)
+    val clazz: Class[_] = implicitly[ClassTag[T]].runtimeClass
+
+    (schemaAsTypedResult, typingResult) match {
+      case (TypedClass(schemaClass, _), TypedClass(typeClass, _)) if typeClass.equals(schemaClass) => valid
+      case (_, TypedClass(typeClass, _)) if clazz.equals(typeClass) => valid
+      case (_, TypedObjectWithValue(underlying, _)) if underlying.klass.equals(clazz) => valid
```

Comment:
Can u change the order to be the same as above to be visible that it is the same case? I mean clazz == underlying.klass